### PR TITLE
feat: initial Intelij plugin

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Thanks to @bradleydwyer 

I have tested it out and added Hermit support so we run FTL with the right env. 

It's not perfect but I think we should just get this in an iterate. 

#2549

### CODE REVIEW

**Review by wesbillman — APPROVED** [review]
Let's go!!! Nice work @bradleydwyer 🥳

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -156,6 +156,22 @@ jobs:
...
+      - name: Init Hermit
+        uses: cashapp/activate-hermit@v1
+        with:
+          cache: true
+      - name: Build Cache
+        uses: ./.github/actions/build-cache
+      - name: Install Java
+        run: java -version
+      - name: Build Intellij Plugin
+        run: just build-intellij-plugin
```
Comment (alecthomas) [inline .github/workflows/ci.yml:174]: How long does this take? Copy pasta the `if` statement from other jobs if this is going to take more than a minute or two, so we can keep the main PR loop fast.

**Comment by alecthomas** [comment]
This is awesome!

Though why do we want Hermit support in this plugin when there's already a Hermit plugin? Does it not interact well or something?

**Review by alecthomas — APPROVED** [review]

**Comment by stuartwdouglas** [comment]
> This is awesome!
> 
> Though why do we want Hermit support in this plugin when there's already a Hermit plugin? Does it not interact well or something?

The Hermit plugin adds Hermit support to a couple of different IDE actions, but if you just run a process directly it won't pick up the correct Hermit environment.

By running the command like through the ToolCustomizer (which the Hermit plugin provides an instance of) it picks up the hermit environment so things work as expected. It would also work with any other plugin that is doing similar things.

**Comment by stuartwdouglas** [comment]
Something I still need to figure out is that if you have multiple projects open only the first one will succeed, as `ftl dev` can only be run once. We likely need to be able to run the server standalone.

**Comment by stuartwdouglas** [comment]
Looks like LSP support is paid only: https://plugins.jetbrains.com/docs/intellij/language-server-protocol.html#supported-ides

Longer term LSP might not be the way to go for intellij support

**Comment by angelozerr** [comment]
If you need a free LSP support for IntelliJ please try https://github.com/redhat-developer/lsp4ij

 * You can see the current LSP support at https://github.com/redhat-developer/lsp4ij/blob/main/docs%2FLSPSupport.md

 * LSP4IJ provides an LSP console to show LSP traces and see the state of language server https://github.com/redhat-developer/lsp4ij/blob/main/docs%2FUserGuide.md

 * If you want to play quickly with LSP4IJ and your LS without developping an IJ please read https://github.com/redhat-developer/lsp4ij/blob/main/docs/UserDefinedLanguageServer.md

 * If you want to integrate your ls in an IJ plugin please read https://github.com/redhat-developer/lsp4ij/blob/main/docs%2FDeveloperGuide.md

You can read an article about LSP4IJ at https://idetools.dev/blog/lsp4ij-announcement/

More and more plygins are using LSP4IJ plese see https://github.com/redhat-developer/lsp4ij?tab=readme-ov-file#requirements

Hope you could be interested with LSP4IJ and don't hesitate to answer me if you have any questions.

**Comment by stuartwdouglas** [comment]
@angelozerr thanks for the tip, I will check it out

**Comment by alecthomas** [comment]
Thanks @angelozerr, appreciate the heads up!

**Pull request merged**